src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Assembly.NativeAot.cs

```diff
@@ -35,6 +35,15 @@ public static Assembly Load(string assemblyString)
             return Load(name);
         }
 
+        // Performance metric to count the number of assemblies
+        // Caching since in NativeAOT, the number will be the same
+        private static uint? s_assemblyCount;
+        internal static uint GetAssemblyCount()
+        {
+            s_assemblyCount ??= (uint)AppDomain.CurrentDomain.GetAssemblies().Length;
+            return (uint)s_assemblyCount;
```

Comment:
```suggestion
        private static uint s_assemblyCount;
        internal static uint GetAssemblyCount()
        {
            if (s_assemblyCount == 0)
                s_assemblyCount = (uint)AppDomain.CurrentDomain.GetAssemblies().Length;
            return s_assemblyCount;
```
`uint?` is `Nullable<uint>`. Lock-free assignment of `Nullable<uint>` like this is not thread-safe.

Reply:
Do we need a memory barrier here? Or is the idea that the threads can race and worst case you re-assign the same length?

Reply:
Yes, the idea is the threads can race to set this and last one wins.